httomo/runner/loader.py

```diff
@@ -197,11 +294,15 @@ def __init__(
         )
 
         angles_arr = self._get_angles()
-        darks_arr, flats_arr = get_darks_flats(darks, flats, comm)
+        darks_arr, flats_arr = get_darks_flats(darks, flats, preview_config)
 
-        dataset: h5py.Dataset = self._h5file[data_path]
+        dataset: h5py.Dataset = self._get_data()
         self._data = FullFileDataSet(
-            data=dataset,
+            data=dataset[
+                self._preview.config.angles.start : self._preview.config.angles.stop,
+                self._preview.config.detector_y.start : self._preview.config.detector_y.stop,
+                self._preview.config.detector_x.start : self._preview.config.detector_x.stop,
+            ],
```

Comment:
I am quite certain this actually performs a full read operation from the file into RAM. The return type of dataset slicing operations `dataset[:, ...]` is a numpy array. If the data doesn't fit in memory, that's a problem. Ideally the data should sit in the file and only be read when a block is required. 

I understand that this probably needs to move offsets into `FullFileDataSet`....

Reply:
I believe this particular instance of accidentally causing a read of the full data is resolved. But another round of reviewing will determine if the issue is gone for good, or if it's sprung up in a new place... :sweat_smile:

Reply:
I wonder if we shouldn't put a test for that. I think it's possible - with an `assert isinstance(loader._data.data, h5py.Dataset)` or something like that. To make sure it's not an `ndarray` - this would also ensure that future refactorings don't accidentally create the same issue again.

Reply:
Thanks for the note, this has been done in [caa9d26](https://github.com/DiamondLightSource/httomo/pull/209/commits/caa9d260b2e1804b36cd58c6d2981e11dd0ef67e)